test/sql/read_gsheet.test

```diff
@@ -58,6 +58,74 @@ BOS	35.5	23.0	Boston MA	Northeast
 BNY	56.0	24.5	Brooklyn NY	Northeast
 BUF	32.5	23.0	Buffalo NY	Northeast
 
+# Test the range parameter
+query II
+FROM read_gsheet('11QdEasMWbETbFVxry-SsD8jVcdYIT1zBQszcF84MdE8', sheet='Sheet1', range='A2:B7', header=false);
+----
+Alice	30.0
+Bob	25.0
+Charlie	45.0
+Drake	NULL
+NULL	NULL
+Archie	99.0
+
+query II
+FROM read_gsheet('11QdEasMWbETbFVxry-SsD8jVcdYIT1zBQszcF84MdE8', sheet='Sheet1', range='A2:B7');
+----
+Bob	25.0
+Charlie	45.0
+Drake	NULL
+NULL	NULL
+Archie	99.0
+
+# Test the range parameter from a quoted sheet
+query II
+FROM read_gsheet('11QdEasMWbETbFVxry-SsD8jVcdYIT1zBQszcF84MdE8', sheet='''Sheet1!''', range='A2:B7');
+----
+Bob	25.0
+Charlie	45.0
+Drake	NULL
+NULL	NULL
+Archie	99.0
+
+# Test the range parameter from a quoted sheet with A1 notation
+query II
+FROM read_gsheet('11QdEasMWbETbFVxry-SsD8jVcdYIT1zBQszcF84MdE8', sheet='''Sheet1!''!A2:B7');
+----
+Bob	25.0
+Charlie	45.0
+Drake	NULL
+NULL	NULL
+Archie	99.0
+
+# Test the range parameter using A1 notation
+query II
+FROM read_gsheet('11QdEasMWbETbFVxry-SsD8jVcdYIT1zBQszcF84MdE8', sheet='Sheet1!A2:B7');
+----
+Bob	25.0
+Charlie	45.0
+Drake	NULL
+NULL	NULL
+Archie	99.0
+
+# Test single value from range
+# NOTE: *must* use `header=false` to avoid uncaught bind error
```

Comment:
Ah i just saw this comment

Reply:
There must be a bug caused by empty results. I'll need to do some debugging to see if maybe the shape of the fetch sheets response is different for single cell ranges or if the error comes from something more simple. Will investigate and put up an issue once I've tracked down the souce